include/ts/apidefs.h.in

```diff
@@ -1119,399 +1531,9 @@ typedef struct TSDbgCtl_s {
-extern tsapi const char *TS_HTTP_METHOD_HEAD;
-extern tsapi const char *TS_HTTP_METHOD_OPTIONS;
-extern tsapi const char *TS_HTTP_METHOD_POST;
-extern tsapi const char *TS_HTTP_METHOD_PURGE;
-extern tsapi const char *TS_HTTP_METHOD_PUT;
-extern tsapi const char *TS_HTTP_METHOD_TRACE;
-extern tsapi const char *TS_HTTP_METHOD_PUSH;
-
-/* --------------------------------------------------------------------------
-   HTTP methods string lengths */
-extern tsapi int TS_HTTP_LEN_CONNECT;
-extern tsapi int TS_HTTP_LEN_DELETE;
-extern tsapi int TS_HTTP_LEN_GET;
-extern tsapi int TS_HTTP_LEN_HEAD;
-extern tsapi int TS_HTTP_LEN_OPTIONS;
-extern tsapi int TS_HTTP_LEN_POST;
-extern tsapi int TS_HTTP_LEN_PURGE;
-extern tsapi int TS_HTTP_LEN_PUT;
-extern tsapi int TS_HTTP_LEN_TRACE;
-extern tsapi int TS_HTTP_LEN_PUSH;
-
-/* --------------------------------------------------------------------------
-   TLS Next Protocol well-known protocol names. */
-
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_0_9;
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_1_0;
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_1_1;
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_2_0;
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_3;
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_3_D29;
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_QUIC;
-extern tsapi const char *const TS_ALPN_PROTOCOL_HTTP_QUIC_D29;
-
-extern tsapi int TS_ALPN_PROTOCOL_INDEX_HTTP_0_9;
-extern tsapi int TS_ALPN_PROTOCOL_INDEX_HTTP_1_0;
-extern tsapi int TS_ALPN_PROTOCOL_INDEX_HTTP_1_1;
-extern tsapi int TS_ALPN_PROTOCOL_INDEX_HTTP_2_0;
-extern tsapi int TS_ALPN_PROTOCOL_INDEX_HTTP_3;
-extern tsapi int TS_ALPN_PROTOCOL_INDEX_HTTP_QUIC;
-
-extern tsapi const char *const TS_ALPN_PROTOCOL_GROUP_HTTP;
-extern tsapi const char *const TS_ALPN_PROTOCOL_GROUP_HTTP2;
-
-extern tsapi const char *const TS_PROTO_TAG_HTTP_1_0;
-extern tsapi const char *const TS_PROTO_TAG_HTTP_1_1;
-extern tsapi const char *const TS_PROTO_TAG_HTTP_2_0;
-extern tsapi const char *const TS_PROTO_TAG_HTTP_3;
-extern tsapi const char *const TS_PROTO_TAG_HTTP_QUIC;
-extern tsapi const char *const TS_PROTO_TAG_TLS_1_3;
-extern tsapi const char *const TS_PROTO_TAG_TLS_1_2;
-extern tsapi const char *const TS_PROTO_TAG_TLS_1_1;
-extern tsapi const char *const TS_PROTO_TAG_TLS_1_0;
-extern tsapi const char *const TS_PROTO_TAG_TCP;
-extern tsapi const char *const TS_PROTO_TAG_UDP;
-extern tsapi const char *const TS_PROTO_TAG_IPV4;
-extern tsapi const char *const TS_PROTO_TAG_IPV6;
-
-/* --------------------------------------------------------------------------
-   MLoc Constants */
-/**
-    Use TS_NULL_MLOC as the parent in calls that require a parent
-    when an TSMLoc does not have a parent TSMLoc. For example if
-    the TSMLoc is obtained by a call to TSHttpTxnClientReqGet(),
-
- */
-#define TS_NULL_MLOC ((TSMLoc)0)
+extern "C" void TSPluginInit(int argc, const char *argv[]);
```

Comment:
Might make a comment that this is "C" because it's looked up by name and not implicitly.

Reply:
See above line 1518.